server/routerlicious/packages/services-utils/src/redisUtils.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Deferred } from "@fluidframework/common-utils";
+import { Redis } from "ioredis";
+
+export interface IRedisParameters {
+    prefix?: string;
+    expireAfterSeconds?: number;
+}
+
+export async function executeRedisMultiWithHmsetExpire(
+    client: Redis,
+    key: string,
+    data: { [key: string]: any },
+    expireAfterSeconds: number) {
+    const deferred = new Deferred<void>();
```

Comment:
Do we need a deferred here? Seems like we can do return new Promise<>((resolve, reject) and just resolve/reject inside?

Reply:
That sounds like a good idea too. But I thought that `Deferred` was the standard way to do it across the project, since it "creates a promise and the ability to resolve or reject it". For example, [we use it in KafkaRunner](https://github.com/microsoft/FluidFramework/blob/cd3c8a8f3419f0dd3f68fba4f73024dfad4a12e8/server/routerlicious/packages/lambdas-driver/src/kafka-service/runner.ts#L36).

If `Deferred` is not the standard way (or at least not recommended in this case), let me know and I can gladly change it.

Reply:
Usually, deferred are more useful for long running promises, where we can't say immediately whether the operation succeeded or not. In this case, we want to execute a bunch of commands and want to return the success/failure immediately. So returning a promise looks better.